tests/test_galaxy_proxy_cache.py

```
@@ -0,0 +1,89 @@
+            tmp_path: Pytest-provided temporary directory.
+        """
+        path = _safe_wheel_path(tmp_path, "ansible_collection-1.0.0-py3-none-any.whl")
+        assert path == tmp_path / "ansible_collection-1.0.0-py3-none-any.whl"
+
+    def test_rejects_dotdot_traversal(self, tmp_path: Path) -> None:
+        """Parent-directory traversal in the filename is rejected.
+
+        Args:
+            tmp_path: Pytest-provided temporary directory.
+        """
+        with pytest.raises(ValueError, match="escapes cache directory|Invalid wheel filename"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)